File: criptografia.py

def contadorString(string):
    cont = 0
    for i in string:
        cont = cont + 1
        
    return cont

entrada = input()


novo = []
for i in range(contadorString(entrada)):
    
    if entrada[i] == ' ':
        pass
    
    aux = ord(entrada[i])
    
    
    if (aux > 64 and aux <= 90) or (aux >= 97 and aux <= 122):
        aux = aux + 3
        novo.append(chr(aux))
    else:  
        novo.append(chr(aux))
        pass
    
for i in novo:
    for j in novo:
        aux2 = novo[j]
        novo[j] = novo[j+1]
        novo[j+1] = aux2
        
        
print(entrada)
print(novo)

File: criptografia.c
#include <stdio.h>


int contadorString(char *s){
    int cont = 0, i = 0;
    while(s[i] != '\0'){
        cont++;
        i++;
    }
    return cont - 1;
}

void inverterString(char *str) {
    
    int comprimento = 0;
    while (str[comprimento] != '\0') {
        comprimento++;
    }

    
    int inicio = 0;
    int fim = comprimento - 2;

    while (inicio < fim) {
        
        char temp = str[inicio];
        str[inicio] = str[fim];
        str[fim] = temp;

        
        inicio++;
        fim--;
    }
}

int main(){
    int n, cont = 0;
    scanf("%d", &n);

    do{
        char entrada[10];
        fgets(entrada, 10, stdin);

        for(int i = 0; i < contadorString(entrada); i++){
            printf("%c", entrada[i]);
        }
        printf("\n");
    

    

    

    //printf("%d", contadorString(entrada));
    //printf("%d", intLetra);
    char novoVetor[contadorString(entrada)];
    for(int i = 0; i < contadorString(entrada); i++){
        if(entrada[i] == ' '){

        }
        int aux = (int)entrada[i];

        if ((aux > 64 && aux <= 90) || (aux >= 97 && aux <= 122)){
            aux = aux + 3;
            novoVetor[i] = aux;
        } else {
            novoVetor[i] = aux;
        }
    }

    inverterString(novoVetor);


    int metade = contadorString(entrada)/2;
    char terceiroVetor[contadorString(entrada)];
    
    for(int i = 0; i < contadorString(entrada); i++){
        int aux = (int)novoVetor[i];

        if(i >= metade){
            aux = aux - 1;
            terceiroVetor[i] = aux;
        } else {
            terceiroVetor[i] = aux;
        }
        
        
    }

    for(int i = 0; i < contadorString(entrada); i++){
        printf("%c", terceiroVetor[i]);
    }
    printf("\n");

    }while(cont < n);
    
}